### PR TITLE
arm64: dts: qcom: msm8916-samsung-j5x: Add initial dts (v5)

### PR DESCRIPTION
**Pull request by wonderfulShrineMaidenOfParadise**
- Update Hall sensor, MUIC and WCNSS
- K2HH Accelerometer
- Modem and sound
- Audio jack

### CODE REVIEW

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts [hunk 1]
```diff
@@ -46,20 +65,43 @@
...
+			/* HACK: Replace with SM5703 MUIC when driver is available */
+			compatible = "siliconmitus,sm5502-muic";
+			reg = <0x25>;
+
+			interrupt-parent = <&msmgpio>;
+			interrupts = <121 IRQ_TYPE_EDGE_FALLING>;
+
+			pinctrl-names = "default";
+			pinctrl-0 = <&muic_int_default>;
+		};
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts:102]: TODO: https://github.com/msm8916-mainline/linux/pull/207#discussion_r816772280

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts [hunk 2]
```diff
@@ -89,6 +89,39 @@
...
+			pinctrl-names = "default";
+			pinctrl-0 = <&accel_int_default>;
+
+			vdd-supply = <&pm8916_l6>;
+			vddio-supply = <&pm8916_l6>;
+
+			st,drdy-int-pin = <1>;
+			mount-matrix = "0", "-1", "0",
+				       "1", "0", "0",
+				       "0", "0", "1";
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts:134]:
```c
			k2hh@1D {
				compatible = "stm,k2hh";
				reg = <0x1d>;
				interrupt-parent = <0x46>;
				interrupts = <0x31 0x00>;
				stm,irq_gpio = <0x46 0x31 0x00>;
				stm,vio-supply = <0x50>;
				stm,axis_map_x = <0x01>;
				stm,axis_map_y = <0x00>;
				stm,axis_map_z = <0x02>;
				stm,negate_x = <0x00>;
				stm,negate_y = <0x01>;
				stm,negate_z = <0x01>;
				stm,poll_interval = <0x64>;
				stm,min_interval = <0x02>;
			};
```

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts [hunk 3]
```diff
@@ -89,6 +89,39 @@
...
+		};
+	};
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts:136]:
```c
			k2hh@1D {
				compatible = "stm,k2hh";
				reg = <0x1d>;
				interrupt-parent = <0x46>;
				interrupts = <0x31 0x00>;
				stm,irq_gpio = <0x46 0x31 0x00>;
				stm,vio-supply = <0x50>;
				stm,axis_map_x = <0x01>;
				stm,axis_map_y = <0x00>;
				stm,axis_map_z = <0x02>;
				stm,negate_x = <0x00>;
				stm,negate_y = <0x01>;
				stm,negate_z = <0x01>;
				stm,poll_interval = <0x64>;
				stm,min_interval = <0x02>;
			};
```

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts [hunk 4]
```diff
@@ -46,20 +77,76 @@
...
+			pinctrl-names = "default";
+			pinctrl-0 = <&accel_int_default>;
+
+			vdd-supply = <&pm8916_l6>;
+			vddio-supply = <&pm8916_l6>;
+
+			st,drdy-int-pin = <1>;
+			mount-matrix = "0", "-1", "0",
+				       "1", "0", "0",
+				       "0", "0", "1";
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts:134]:
```suggestion
				       "0", "0", "-1";
```

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts [hunk 5]
```diff
@@ -155,6 +167,14 @@
 	cd-gpios = <&msmgpio 38 GPIO_ACTIVE_LOW>;
 };
 
+&sound {
+	status = "okay";
+	audio-routing =
+		"AMIC1", "MIC BIAS External1",
+		"AMIC2", "MIC BIAS Internal2",
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts:174]: Have you tested if this works properly (tried recording audio from headset microphone)?

This seems related to https://github.com/msm8916-mainline/linux/pull/189/files#r621162048 downstream, perhaps we can just assume it's same on j5 2015 and 2016.

**Review by stephan-gh — CHANGES_REQUESTED** [review]

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts (hunk 5 above)
Comment (wonderfulShrineMaidenOfParadise) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts:174]: Headset microphone is broken with this. Trying to fix it.

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts [hunk 6]
```diff
@@ -80,14 +167,31 @@
...
+};
+
+&wcd_codec {
+	jack-gpios = <&msmgpio 110 GPIO_ACTIVE_LOW>;
+	qcom,micbias-lvl = <2800>;
+	qcom,mbhc-vthreshold-low = <75 150 237 450 500>;
+	qcom,mbhc-vthreshold-high = <75 150 237 450 500>;
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&jack_default>;
```
Comment (wonderfulShrineMaidenOfParadise) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts:194]: Headphone buttons are broken as well.

File: arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts (hunk 5 above)
Comment (wonderfulShrineMaidenOfParadise) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-j5x.dts:174]: Same commit works on j3ltezt. However there are things to figure out:
- Is my j5x jack broken?
- Are jacks on j5x and j3ltezt similar?

**Comment by stephan-gh** [comment]
Pushed to rebase as-is for now, if you figure out what's wrong with the headset mic (or maybe it's really just your device?), then feel free to push additional squash!/fixup! commits.

**Pull request closed without merging**